src/ComposerRequireChecker/Cli/ResultsWriter/JsonFile.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ComposerRequireChecker\Cli\ResultsWriter;
+
+use DateTimeImmutable;
+use RuntimeException;
+
+use function dirname;
+use function file_put_contents;
+use function function_exists;
+use function is_dir;
+use function json_encode;
+use function mkdir;
+use function sprintf;
+
+final class JsonFile implements ResultsWriter
+{
+    private string $filePathname;
+    private string $applicationVersion;
+
+    public function __construct(string $filePathname, string $applicationVersion)
+    {
+        if (! function_exists('json_encode')) {
```

Comment:
We should add `ext-json` as a dependency: after all, this library is called `composer-require-checker`, and we should eat our own dogfood :)

Reply:
It’s really only there for PHP 7.4. Since in a phar we do not check dependencies first. Yes in composer.json we rely on ext-json already.

Reply:
Doesn't the dumped autoloader (that we throw at the `.phar` afterwards) include these checks already?

Reply:
I’ll have a look in the generated phar and which checks it might do. But probably it’s enough to let php crash and point to composer.json as reason.

Reply:
BTW, ext-json is bundled with PHP since 7.3, IIRC

Reply:
Yes, but in custom builds it can be disabled until 8.0 where the compile flag “—disable-json” is removed.

Reply:
TBH, who runs with `--disable-json` probably deserves a crash :P